test/integrations/destinations/linkedin_ads/dataDelivery/oauth.ts

```diff
@@ -0,0 +1,218 @@
+      conversion: 'urn:lla:llaPartnerConversion:23456',
+      conversionHappenedAt: 1697241600000,
+      conversionValue: {
+        amount: '0',
+        currencyCode: 'USD',
+      },
+      eventId: 'a80f82be-9bdc-4a9f-b2a5-15621ee41df8',
+      user: {
+        userIds: [
+          {
+            idType: 'SHA256_EMAIL',
+            idValue: 'abc@gmail.com',
+          },
+        ],
+      },
+    },
+  ],
+};
+export const statTags = {
+  destType: 'LINKEDIN_ADS',
+  errorCategory: 'network',
+  destinationId: 'default-destinationId',
+  workspaceId: 'default-workspaceId',
+  errorType: 'aborted',
+  feature: 'dataDelivery',
+  implementation: 'native',
+  module: 'destination',
+};
+
+export const metadata = {
+  jobId: 1,
+  attemptNum: 1,
+  userId: 'default-userId',
+  destinationId: 'default-destinationId',
+  workspaceId: 'default-workspaceId',
+  sourceId: 'default-sourceId',
+  secret: {
+    accessToken: 'default-accessToken',
+  },
+  dontBatch: false,
+};
+
+export const headerBlockWithCorrectAccessToken = {
+  Authorization: 'Bearer default-accessToken',
+  'Content-Type': 'application/json',
+  'LinkedIn-Version': '202402',
+  'X-RestLi-Method': 'BATCH_CREATE',
+  'X-Restli-Protocol-Version': '2.0.0',
+};
+
+const commonRequestParameters = {
+  headers: headerBlockWithCorrectAccessToken,
+  JSON: testJSONData,
+};
+const commonRequestParametersWithInvalidAccess = {
+  headers: { ...headerBlockWithCorrectAccessToken, Authorization: 'Bearer invalidToken' },
+  JSON: testJSONData,
+  accessToken: 'invalidToken',
+};
+
+const commonRequestParametersWithRevokedAccess = {
+  headers: { ...headerBlockWithCorrectAccessToken, Authorization: 'Bearer revokedToken' },
+  JSON: testJSONData,
+  accessToken: 'revokedToken',
+};
+
+export const oauthScenariosV1: ProxyV1TestData[] = [
+  {
+    id: 'linkedin_ads_v1_oauth_scenario_1',
+    name: 'linkedin_ads',
+    description: 'app event fails due to revoked access token error',
+    successCriteria: 'Should return 500 with revoked access token error',
+    scenario: 'Business',
+    feature: 'dataDelivery',
+    module: 'destination',
+    version: 'v1',
+    input: {
+      request: {
+        body: generateProxyV1Payload({
+          endpoint: `https://api.linkedin.com/rest/conversionEvents`,
+          ...commonRequestParametersWithRevokedAccess,
+        }),
+        method: 'POST',
+      },
+    },
+    output: {
+      response: {
+        status: 500,
+        body: {
+          output: {
+            response: [
+              {
+                error:
+                  '{"status":401,"serviceErrorCode":65601,"code":"REVOKED_ACCESS_TOKEN","message":"The token used in the request has been revoked by the user"}',
+                statusCode: 500,
+                metadata: { ...metadata, secret: { accessToken: 'revokedToken' } },
+              },
+            ],
+            statTags: { ...statTags, errorType: 'retryable' },
+            authErrorCategory: 'REFRESH_TOKEN',
```

Comment:
Why are we refreshing for this?

Reply:
`REVOKED_ACCESS_TOKEN` is not for permission issue actually. Anytime if a new access token is created, but still we try to send data with older token in between, it will provide us with revoked access token error. In that case we should retry right?

Reply:
I did not get it clearly Refresh should be done when access token is expired and revoke means even after refreshing it wont work

Reply:
So always there will be a single valid access token ?
Do we have any doc around this error(`REVOKED_ACCESS__TOKEN`) ?

Reply:
But here if we refreshing it is working

Reply:
@sanpj2292 no, I do not find any specific doc based on this, but I can show you by calling the native APIs

Reply:
discussed with @sanpj2292 and made it aborted